src/AppBundle/Markdown/UrlParser.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace AppBundle\Markdown;
+
+use League\CommonMark\Inline\Element\Link;
+use League\CommonMark\Inline\Parser\AbstractInlineParser;
+use League\CommonMark\InlineParserContext;
+
+/**
+ * @author David Badura <d.a.badura@gmail.com>
+ */
+class UrlParser extends AbstractInlineParser
+{
+    const PATTERN = '#(?i)\b((?:[a-z][\w-]+:(?:/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:\'".,<>?«»“”‘’]))#iS';
```

Comment:
@DavidBadura that's easy to understand and comprehensive :wink: I guess you didn't write that regular expression yourself? Would be an excellent point to introduce some unit tests, as those tests would illustrate which URLs are detected and which are not, e.g. is file:///... supported? etc.